docs/content/connectors/mongodb-cdc.md

```diff
@@ -0,0 +1,440 @@
+
+Setup MongoDB
+----------------
+
+### Availability
+- MongoDB version
+
+  MongoDB version >= 3.6 <br>
+We use [change streams](https://docs.mongodb.com/manual/changeStreams/) feature (new in version 3.6) to capture change data.
+
+- Cluster Deployment
+
+  [replica sets](https://docs.mongodb.com/manual/replication/) or [sharded clusters](https://docs.mongodb.com/manual/sharding/) is required.
+
+- Storage Engine
+
+  [WiredTiger](https://docs.mongodb.com/manual/core/wiredtiger/#std-label-storage-wiredtiger) storage engine is required.
+
+- [Replica set protocol version](https://docs.mongodb.com/manual/reference/replica-configuration/#mongodb-rsconf-rsconf.protocolVersion)
+
+  Replica set protocol version 1 [(pv1)](https://docs.mongodb.com/manual/reference/replica-configuration/#mongodb-rsconf-rsconf.protocolVersion) is required. <br>
+Starting in version 4.0, MongoDB only supports pv1. pv1 is the default for all new replica sets created with MongoDB 3.2 or later.
+
+- Privileges
+
+  `changeStream` and `read` privileges are required by MongoDB Kafka Connector. 
+
+  You can use the following example for simple authorization.<br>
+  For more detailed authorization, please refer to [MongoDB Database User Roles](https://docs.mongodb.com/manual/reference/built-in-roles/#database-user-roles).
+
+  ```javascript
+  use admin;
+  db.createUser({
+    user: "flinkuser",
+    pwd: "flinkpw",
+    roles: [
+      { role: "read", db: "admin" }, //read role includes changeStream privilege 
+      { role: "readAnyDatabase", db: "admin" } //for snapshot reading
+    ]
+  });
+  ```
+
+
+How to create a MongoDB CDC table
+----------------
+
+The MongoDB CDC table can be defined as following:
+
+```sql
+-- register a MongoDB table 'products' in Flink SQL
+CREATE TABLE products (
+  _id STRING, // must be declared
+  name STRING,
+  weight DECIMAL(10,3),
+  tags ARRAY<STRING>, -- array
+  price ROW<amount DECIMAL(10,2), currency STRING>, -- embedded document
+  suppliers ARRAY<ROW<name STRING, address STRING>>, -- embedded documents
+  PRIMARY KEY(_id) NOT ENFORCED
+) WITH (
+  'connector' = 'mongodb-cdc',
+  'hosts' = 'localhost:27017,localhost:27018,localhost:27019',
+  'username' = 'flinkuser',
+  'password' = 'flinkpw',
+  'database' = 'inventory',
+  'collection' = 'products'
+);
+
+-- read snapshot and change events from products collection
+SELECT * FROM products;
+```
+
+**Note that** 
+
+MongoDB's change event record doesn't have update before message. So, we can only convert it to Flink's UPSERT changelog stream.
+An upsert stream requires a unique key, so we must declare `_id` as primary key. 
+We can't declare other column as primary key, becauce delete operation do not contain's the key and value besides `_id` and `sharding key`.
+
+Connector Options
+----------------
```

Comment:
From the documentation, it seems that we don't support `debezium.*` options for mongodb-cdc. Should we use `helper.validate();` instead of `helper.validateExcept(DebeziumOptions.DEBEZIUM_OPTIONS_PREFIX);` at [this line](https://github.com/ververica/flink-cdc-connectors/blob/master/flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/table/MongoDBTableSourceFactory.java#L180)?

Reply:
Thanks @wuchong 

The original intention I reserved it was to reserve the configuration for the [debezium engine properties](https://debezium.io/documentation/reference/1.5/development/engine.html#engine-properties).

It seems we don't need to it, I'll open a new PR to fix it.

Reply:
Fixed in #420

Reply:
Yes, as you said we didn't use debezium mongodb connector.